contracts/src/v0.8/l2ep/test/v1_0_0/shared/BaseSequencerUptimeFeed.t.sol

```diff
@@ -0,0 +1,356 @@
+    assertEq(roundId, 1);
+    assertEq(answer, 0);
+    assertEq(answeredInRound, roundId);
+    assertEq(startedAt, updatedAt);
+
+    // Assert latestRoundData corresponds to latest round id
+    (roundId, answer, startedAt, updatedAt, answeredInRound) = s_sequencerUptimeFeed.latestRoundData();
+    assertEq(roundId2, roundId);
+    assertEq(answer2, answer);
+    assertEq(startedAt2, startedAt);
+    assertEq(updatedAt2, updatedAt);
+    assertEq(answeredInRound2, answeredInRound);
+  }
+
+  /// @notice it should revert from #getRoundData when round does not yet exist (future roundId)
+  function test_RevertGetRoundDataWhenRoundDoesNotExistYet() public {
+    // Sets msg.sender and tx.origin to a valid address
+    vm.startPrank(s_l1OwnerAddr, s_l1OwnerAddr);
+
+    // Gets data from a round that has not happened yet
+    vm.expectRevert(BaseSequencerUptimeFeed.NoDataPresent.selector);
+    s_sequencerUptimeFeed.getRoundData(2);
+  }
+
+  /// @notice it should return the #getAnswer for the latest round
+  function test_GetValidAnswer() public {
+    // Sets msg.sender and tx.origin to a valid address
+    vm.startPrank(s_aliasedL1OwnerAddress, s_aliasedL1OwnerAddress);
+
+    uint256 startedAt;
+    (, , startedAt, , ) = s_sequencerUptimeFeed.latestRoundData();
+
+    s_sequencerUptimeFeed.updateStatus(true, uint64(startedAt + 1000));
+
+    assertEq(0, s_sequencerUptimeFeed.getAnswer(1));
+  }
+
+  /// @notice it should revert from #getAnswer when round does not yet exist (future roundId)
+  function test_RevertGetAnswerWhenRoundDoesNotExistYet() public {
+    // Sets msg.sender and tx.origin to a valid address
+    vm.startPrank(s_l1OwnerAddr, s_l1OwnerAddr);
+
+    // Gets data from a round that has not happened yet
+    vm.expectRevert(BaseSequencerUptimeFeed.NoDataPresent.selector);
+    s_sequencerUptimeFeed.getAnswer(2);
+  }
+
+  /// @notice it should return the #getTimestamp for the latest round
+  function test_GetValidTimestamp() public {
+    // Sets msg.sender and tx.origin to a valid address
+    vm.startPrank(s_aliasedL1OwnerAddress, s_aliasedL1OwnerAddress);
+
+    uint256 startedAt;
+    (, , startedAt, , ) = s_sequencerUptimeFeed.latestRoundData();
+
+    s_sequencerUptimeFeed.updateStatus(true, uint64(startedAt + 1000));
+
+    assertEq(startedAt, s_sequencerUptimeFeed.getTimestamp(1));
+  }
+
+  /// @notice it should revert from #getTimestamp when round does not yet exist (future roundId)
+  function test_RevertGetTimestampWhenRoundDoesNotExistYet() public {
+    // Sets msg.sender and tx.origin to a valid address
+    vm.startPrank(s_l1OwnerAddr, s_l1OwnerAddr);
+
+    // Gets data from a round that has not happened yet
+    vm.expectRevert(BaseSequencerUptimeFeed.NoDataPresent.selector);
+    s_sequencerUptimeFeed.getTimestamp(2);
+  }
+}
+
+contract BaseSequencerUptimeFeed_ProtectReadsOnAggregatorV2V3InterfaceFunctions is BaseSequencerUptimeFeedTest {
```

Comment:
Contracts should be named `<contract_name>_<function_being_tested>`

Reply:
Fixed!